azure/compute/src/main/java/org/jclouds/azurecompute/compute/functions/OSImageToImage.java

```diff
@@ -0,0 +1,88 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file has been modified by Microsoft Open Technologies, Inc.
+ * Modifications copyright (c) Microsoft Open Technologies, Inc.
+ */
+package org.jclouds.azurecompute.compute.functions;
+
+import java.util.Map;
+
+import javax.annotation.Resource;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.azurecompute.domain.OSImage;
+import org.jclouds.compute.domain.Image;
+import org.jclouds.compute.domain.ImageBuilder;
+import org.jclouds.compute.domain.OperatingSystem;
+import org.jclouds.compute.domain.OsFamily;
+import org.jclouds.compute.domain.Image.Status;
+import org.jclouds.compute.reference.ComputeServiceConstants;
+import org.jclouds.logging.Logger;
+
+import com.google.common.base.Function;
+
+@Singleton
+public class OSImageToImage implements Function<OSImage, Image> {
+    private static String PUBLIC_CATEGORY = "Public";
+    @Resource
+    @Named(ComputeServiceConstants.COMPUTE_LOGGER)
+    protected Logger logger = Logger.NULL;
+
+    private final Map<OsFamily, Map<String, String>> osVersionMap;
+
+    @Inject
+    OSImageToImage(Map<OsFamily, Map<String, String>> osVersionMap) {
+        this.osVersionMap = osVersionMap;
+    }
+
+    protected OperatingSystem parseOs(OSImage from) {
+        OsFamily osFamily = null;
+        String osName = from.getImageFamily();  //from.getLabel();
+        String osArch = ""; //from.getImageFamily();
+        String osVersion = from.getPublishedDate() == null ? null : "" + from.getPublishedDate().getTimeInMillis();
+        String osDescription = ""; //from.getDescription() == null ? "" : from.getDescription();
+        try {
+            osFamily = OsFamily.fromValue(from.getOSType().toLowerCase());
+        } catch (IllegalArgumentException e) {
+            logger.debug("<< didn't match os(%s)", from.getName());
+        }
+        boolean is64Bit = true;
+//                (from.getName() != null && from.getName().contains("64")) ||
+//                (from.getDescription() != null && from.getDescription().contains("64")) ||
+//                (from.getLabel() != null && from.getLabel().contains("64")) || osFamily == OsFamily.WINDOWS;
+
+        return new OperatingSystem(osFamily, osName, osVersion, osArch, osDescription, is64Bit);
+    }
+
+    @Override
+    public Image apply(OSImage from) {
+        ImageBuilder builder = new ImageBuilder();
+        builder.ids(from.getName());
```

Comment:
This sets two fields: the `id` and the `providerId`. The former is the ID jclouds will use to identify uniquely the image across all regions, and the latter the *real* ID of the resource in the provider. Is this true in this case?
Usually jclouds uses a different ID and prepends the region to the provider ID. Take this into accound to make sure the providerId and the id (which is the one used when calling getImage(id) in the computeservice) are configured properly.